nixos/modules/services/web-servers/apache-httpd/default.nix

```diff
@@ -641,10 +643,10 @@ in
         '';
       }
     ] ++ map (name: mkCertOwnershipAssertion {
-      inherit (cfg) group user;
       cert = config.security.acme.certs.${name};
       groups = config.users.groups;
-    }) dependentCertNames;
+      services = [ config.systemd.services.httpd ] ++ lib.optional (vhostCertNames != []) config.systemd.services.httpd-config-reload;
```

Comment:
httpd-config-reload only needs permission for the certs because we do the config test in [ExecStartPre](https://github.com/NixOS/nixpkgs/pull/336412/files#diff-c4e76be6d8d687a02103da358697dd78a98738f3a1bc20c52b3a00fbccf2e6a9R813).

What we could do instead is change the ExecReload entry for httpd itself to run the config test followed by the actual reload, by providing a list of commands (Which get expanded into multiple ExecReload statements). Then we can remove the change on this line.

Reply:
Yes that seems good to me. I didn't want to change those kinds of things in this PR.  
I'll open a fix PR today.

I think we can even remove the reload services altogether but that's for later.

Reply:
I made a quick stab at this this evening, and ran the acme test suite against it. Feel free to cherry-pick and open a PR :) https://github.com/NixOS/nixpkgs/compare/master...m1cr0man:nixpkgs:acme-assertion-fix

Reply:
Thanks! I also have a local change to fix the assert code itself.  
Unfortunately, there's no easy way to test asserts since the NixOS test config must build to be ran as a VM. I did play with a workaround, but not sure if it's worth doing that now. I'll think about it more and make a PR later.

Reply:
I opened a simple PR to fix just the assert message: https://github.com/NixOS/nixpkgs/pull/362271

I didn't take your commit, instead made a PR to go a step further and remove the `*-config-reload` services: https://github.com/NixOS/nixpkgs/pull/362296